### PR TITLE
add armv7-a and armv7-r targets

### PR DESCRIPTION
**Pull request by domin144**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -966,7 +966,7 @@ add_library_variant(
 add_library_variant(
     armv6m
     SUFFIX soft_nofp
-    COMPILE_FLAGS "-mfloat-abi=soft -march=armv6m"
+    COMPILE_FLAGS "-mfloat-abi=soft -march=armv6-m"
```
Comment (mplatings) [inline CMakeLists.txt:969]: I'd prefer these changes weren't made. `armv6m` works with both `-march` and `--target` whereas `armv6-m` only works with the former.

File: CMakeLists.txt [hunk 2]
```diff
@@ -977,10 +977,78 @@ add_library_variant(
     RAM_SIZE 2M
     STACK_SIZE 4K
 )
+add_library_variant(
+    armv7a
+    SUFFIX soft_nofp
+    COMPILE_FLAGS "-mfloat-abi=soft -march=armv7-a+nofp"
+    MULTILIB_FLAGS "--target=thumbv7-none-unknown-eabi -mfpu=none"
```
Comment (mplatings) [inline CMakeLists.txt:984]: I'm pretty sure this is incorrect. Please add tests in `test/multilib` to ensure these work and continue working.

**Review by mplatings — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (domin144) [reply CMakeLists.txt]: Ok. Reverting. The motivation was to be in line with GCC documentation.

File: CMakeLists.txt (hunk 2 above)
Comment (domin144) [reply CMakeLists.txt]: You are right.
I added tests and fixed the flags. Thanks.

**Review by mplatings — APPROVED** [review]

**Pull request merged**